UIPS/UIP-0131.md

```diff
@@ -0,0 +1,121 @@
+
+### State Management
+
+The Embassy contract maintains just the current state of points on its chain---their ownership addresses and proxy configurations. When a point arrives through the Ministry bridge, the Embassy registers only this essential current state data.
+
+For returns, the oracle-signed tickets need only contain the point's current ownership address and proxy configurations. This lightweight approach leverages the oracle trust model---since oracles stake their galaxies on providing accurate state transitions, maintaining historical state becomes unnecessary. The optimistic rollup's challenge mechanism relies on oracle signatures rather than state history verification.
+
+### Oracle Security Model
+
+Galaxies secure Ministry by staking their own identities in the contract. To operate as an oracle, a galaxy transfers itself to the Ministry contract as collateral. This ensures stake value inherently exceeds typical transfer values, as galaxy ownership is generally more valuable than subordinate points. While certain points may accumulate exceptional value through smart contract integrations or attached assets that could theoretically exceed a galaxy's worth, this economic reality does not impair the protocol's security model.
+
+The system enforces correct oracle behavior through social consensus and economic incentives. Invalid transfer tickets serve as evidence of misbehavior---any party can submit transaction references showing fraudulent ticket validation to the Ministry contract. These references locate the original transfer records on each chain. The reports support Senate governance decisions which may result in galaxy forfeiture. Multiple oracles provide redundancy in signature collection and transfer validation, ensuring the bridge remains operational even if some oracles are temporarily offline.
+
+The protocol implements rate limiting, mandatory cooldowns, and minimum confirmation periods to prevent timing attacks. A monitoring client watches both chains for fraudulent tickets and automates violation reporting.
+
+TODO: Specify signature thresholds, timeouts, rate limits, and violation report format.
+
+### Arvo Integration
+
+The introduction of multi-chain state requires significant changes to Urbit's PKI verification system. Currently, %jael maintains Azimuth state by synthesizing data from Ethereum and the naive rollup. The addition of Base requires extending this system to handle three distinct sources of truth.
+
+The %base-watcher agent extends %jael's PKI verification capabilities, monitoring Base chain activity and updating Azimuth state accordingly. While the Ministry bridge protocol handles transfer sequencing and validation through its vault mechanism and oracle system, %jael remains responsible for maintaining a consistent view of the PKI state. This includes tracking point locations across chains and handling state rollbacks that may occur before bridge transfers are finalized.
```

Comment:
> %jael remains responsible for maintaining a consistent view of the PKI state. This includes tracking point locations across chains and handling state rollbacks that may occur before bridge transfers are finalized.

As per my comment above, jael presently has no knowledge _at all_ of chain-specific details. I would object to putting that kind of logic in the kernel at this point in time, and _especially_ if that logic has to relate to an ever-expanding set of sources.